connector/subscription_manager.go

```diff
@@ -18,41 +18,46 @@ import (
 
 // SubscriptionManager is responsible for subscriptions management.
 type SubscriptionManager interface {
-	// Add adds a subscription for the provided topic.
-	Add(topic string)
+	// Add adds a subscription for the provided topic. Returns true if it doesn't exist and was successfully subscribed.
+	Add(topic string) bool
 
-	// Remove removes a subscription for the provided topic.
-	Remove(topic string)
+	// Remove removes a subscription for the provided topic. Returns true if it does exist and was successfully unsubscribed.
+	Remove(topic string) bool
 
 	// ForwardTo specifies the connection to which should be the messages forwarded.
 	ForwardTo(conn *MQTTConnection)
 }
 
-type subscriptionmanager struct {
+type subscriptionManager struct {
 	topics sync.Map
 
 	conn atomic.Value
 }
 
-func (m *subscriptionmanager) Add(topic string) {
+func (m *subscriptionManager) Add(topic string) bool {
 	if _, ok := m.topics.LoadOrStore(topic, true); !ok {
 		if connRef := m.conn.Load(); connRef != nil {
```

Comment:
We need "else" here to clean up the topics map. The topic is stored but there is no subscription.

Reply:
This cannot ever happen in the application.